src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+                var outputItemType = OnnxUtils.OnnxToMlNetType(outputNodeInfo.Type);
+                _outputColType = new VectorType(outputItemType, dims);
+                _outputColName = _parent.Outputs[0];// outputNodeInfo.Name;
+                _outputItemRawType = outputItemType.RawType;
+                _host.Assert(_outputItemRawType == _outputColType.ItemType.RawType);
+            }
+
+            public RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var info = new RowMapperColumnInfo[1];
+                info[0] = new RowMapperColumnInfo(_outputColName, _outputColType, null);
+                return info;
+            }
+
+            public Func<int, bool> GetDependencies(Func<int, bool> activeOutput)
+            {
+                return col => activeOutput(0) && (_idvToTensorAdapter.IdvColumnIndex == col);
+            }
+
+            public void Save(ModelSaveContext ctx)
+            {
+                _parent.Save(ctx);
+            }
+
+            public Delegate[] CreateGetters(IRow input, Func<int, bool> activeOutput, out Action disposer)
+            {
+                disposer = null;
+                var getters = new Delegate[1];
+                using (var ch = _host.Start("CreateGetters"))
+                {
+                    if (activeOutput(0))
+                        getters[0] = Utils.MarshalInvoke(MakeGetter<int>, _outputItemRawType, input);
+
+                    ch.Done();
+                    return getters;
+                }
+            }
+
+            private Delegate MakeGetter<T>(IRow input)
+            {
+                _host.AssertValue(input);
+                _host.Assert(typeof(T) == _outputItemRawType);
+
+                _idvToTensorAdapter.InitializeValueGetters(input);
+
+                ValueGetter<VBuffer<T>> valuegetter = (ref VBuffer<T> dst) =>
+                {
+                    var inputTensors = new List<Tensor> { _idvToTensorAdapter.GetTensor() };
+                    var outputTensors = _parent._model.Run(inputTensors);
+                    Contracts.Assert(outputTensors.Count() > 0);
+
+                    var values = dst.Values;
+                    if (Utils.Size(values) < _outputColType.VectorSize)
+                        values = new T[_outputColType.VectorSize];
+
+                    OnnxUtils.CopyTo(outputTensors[0], values);
+                    dst = new VBuffer<T>(values.Length, values, dst.Indices);
+                };
+
+                return valuegetter;
+            }
+        }
+
+        /// <summary>
+        /// IdvToTensorAdapter adapts an Idv (row-iterator interface) to a tensor-iterator interface.
+        /// For an Idv, you'd need to create a cursor and iterate over it to get each rows of the Idv.
+        /// After adaptation, you'd call GetTensor() on the IdvToTensorAdapter object to get the Tensor equivalent of
+        /// each row.
+        /// </summary>
+        internal sealed class IdvToTensorAdapter
```

Comment:
>IdvToTensorAdapter [](start = 30, length = 18)

I would move it to OnnxUtils

Reply:
Moved to OnnxUtils.

---
In reply to: [219613978](https://github.com/dotnet/machinelearning/pull/942#discussion_r219613978) [](ancestors = 219613978)